src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -106,14 +112,15 @@ private object GetFirstOrDefaultValue(OptionKey optionKey, IEnumerable<RoamingPr
             return optionKey.Option.DefaultValue;
         }
 
+        public Task PrefetchAsync(CancellationToken cancellationToken)
+        {
+            // this doesn't require UI thread, doesn't need to prefetch anything.
+            return Task.CompletedTask;
+        }
+
         public bool TryFetch(OptionKey optionKey, out object value)
         {
-            if (_settingManager == null)
-            {
-                Debug.Fail("Manager field is unexpectedly null.");
-                value = null;
-                return false;
-            }
+            var service = _serviceInitializer.GetService(CancellationToken.None);
```

Comment:
if fetching options can be expensive, can we convey that by having TryFetch be cancellable?

Reply:
Alternatively, if we want IOptionPersister to present this synchronous API, can we instead introduce some new provider interface that one uses to get the persister?  then one can to "IOperationPersisterProvider.GetPersisterAsync() ...   persister.TryFetch".  That way we don't have raw, expensive, blocking happening in operations that look like they should be lightweight.

Reply:
we have like 10 or more VS services to move from GetService to GetServiceAsync and if possible to BG, I happen to pick IOptionPersister to try multiple approaches on how we should approach these.

so, like changing IOptionPersister to Async or change TryFetch to async is not the point, 

sure, we might take breaking change and change all Option API to async, but still need to figure out what to do for some services we can't either move BG or Async.

Reply:
> we have like 10 or more VS services to move from GetService to GetServiceAsync and if possible to BG, I happen to pick IOptionPersister to try multiple approaches on how we should approach these.

Sure... i'm just saying that this approach doesn't seem as good as other potential ones we could take.  It mixes concepts in interfaces and makes for a more confusing system (from my perspective).